src/handlers/CrossMarginHandler02.sol

```diff
@@ -0,0 +1,572 @@
+  /// @param _orderIndex Index of the order to cancel.
+  function cancelWithdrawOrder(uint8 _subAccountId, uint256 _orderIndex) external nonReentrant {
+    _cancelWithdrawOrder(msg.sender, _subAccountId, _orderIndex, msg.sender);
+  }
+
+  /**
+   * Internals
+   */
+
+  function _executeOrder(
+    ExecuteOrderVars memory vars,
+    address _account,
+    uint8 _subAccountId,
+    uint256 _orderIndex,
+    bool _isRevert
+  ) internal returns (uint256 _totalFeeReceiver) {
+    vars.subAccount = HMXLib.getSubAccount(_account, _subAccountId);
+    vars.order = withdrawOrders[vars.subAccount][_orderIndex];
+    vars.orderIndex = _orderIndex;
+
+    // Check if this order still exists
+    if (vars.order.account == address(0)) revert ICrossMarginHandler02_NonExistentOrder();
+
+    // ignore this order, if amount = 0
+    if (vars.order.amount == 0) return 0;
+
+    // try executing order
+    try this.executeWithdrawOrder(vars.order) {
+      _handleOrderSuccess(vars.subAccount, _orderIndex);
+    } catch Error(string memory errMsg) {
+      _handleOrderFail(vars, bytes(errMsg), _isRevert);
+    } catch Panic(uint /*errorCode*/) {
+      _handleOrderFail(vars, bytes("Panic occurred while executing the withdraw order"), _isRevert);
+    } catch (bytes memory errMsg) {
+      _handleOrderFail(vars, errMsg, _isRevert);
+    }
+
+    _totalFeeReceiver = vars.order.executionFee;
+  }
+
+  function _cancelWithdrawOrder(
+    address _account,
+    uint8 _subAccountId,
+    uint256 _orderIndex,
+    address _feeReceiver
+  ) internal {
+    address subAccount = HMXLib.getSubAccount(_account, _subAccountId);
+    // SLOAD
+    WithdrawOrder memory _order = withdrawOrders[subAccount][_orderIndex];
+    // Check if this order still exists
+    if (_order.account == address(0)) revert ICrossMarginHandler02_NonExistentOrder();
+    // validate if msg.sender is not owned the order, then revert
+    if (msg.sender != _order.account && !orderExecutors[msg.sender]) revert ICrossMarginHandler02_NotOrderOwner();
+
+    _removeOrder(subAccount, _orderIndex);
+
+    // refund the _order.executionFee
+    _transferOutETH(_order.executionFee, _feeReceiver);
+
+    emit LogCancelWithdrawOrder(
+      payable(_order.account),
+      _order.subAccountId,
+      _orderIndex,
+      _order.token,
+      _order.amount,
+      _order.executionFee,
+      _order.shouldUnwrap
+    );
+
+    delete withdrawOrders[subAccount][_orderIndex];
+  }
+
+  function _addOrder(WithdrawOrder memory _order, address _subAccount, uint256 _orderIndex) internal {
+    withdrawOrdersIndex[_subAccount] = _orderIndex + 1;
+    withdrawOrders[_subAccount][_orderIndex] = _order;
+
+    uint256 _pointer = _encodePointer(_subAccount, uint96(_orderIndex));
+    _activeOrderPointers.add(_pointer);
+    _subAccountActiveOrderPointers[_subAccount].add(_pointer);
+  }
+
+  function _handleOrderSuccess(address _subAccount, uint256 _orderIndex) internal {
```

Comment:
missing the part where feeReceiver should receive gas from success order

please add test case where we assert gas received at feeReceiver.